lib/tokenize.cpp

```diff
@@ -7615,7 +7615,24 @@ void Tokenizer::findGarbageCode() const
     for (const Token *tok = tokens(); tok; tok = tok->next()) {
         if (tok->str() == "{")
             tok = tok->link();
-        else if (tok->isKeyword() && nonGlobalKeywords.count(tok->str()) && !Token::Match(tok->tokAt(-2), "operator %str%"))
+        else if (tok->isKeyword() && Token::simpleMatch(tok->previous(), ") try")) { // function-try-block
+            const Token* endTry = tok->next();
+            // skip initializer list :
+            while (Token::Match(endTry, "[:,] %name% (|{"))
+                endTry = endTry->linkAt(2)->next();
+            if (!Token::simpleMatch(endTry, "{")) {
+                if (endTry)
+                    syntaxError(endTry, "Unexpected token '" + endTry->str() + "'");
+                else
+                    syntaxError(nullptr);
```

Comment:
pass `tok` to syntaxError and write something like `Invalid function-try-catch block code. Did not find '{' for try body.`